pkg/agent/manager/cache/bundle_cache.go

```diff
@@ -2,20 +2,21 @@ package cache
 
 import (
 	"github.com/imkira/go-observer"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
 )
 
 type BundleCache struct {
-	trustDomainID string
-	bundles       observer.Property
+	trustDomain spiffeid.TrustDomain
+	bundles     observer.Property
 }
 
-func NewBundleCache(trustDomainID string, bundle *Bundle) *BundleCache {
+func NewBundleCache(trustDomain spiffeid.TrustDomain, bundle *Bundle) *BundleCache {
 	bundles := map[string]*Bundle{
```

Comment:
I think we should probably just change this to a `map[spiffeid.TrustDomain]*Bundle`

Reply:
Similarly, `BundleStream` and its consumers should be updated to work with the `map[spiffeid.TrustDomain]*Bundle` type

Reply:
👍